internal/component/prometheus/remote/queue/filequeue/filequeue.go

```diff
@@ -0,0 +1,182 @@
+package filequeue
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+	"sort"
+	"strconv"
+	"strings"
+
+	"github.com/go-kit/log"
+	"github.com/grafana/alloy/internal/component/prometheus/remote/queue/types"
+	"github.com/grafana/alloy/internal/runtime/logging/level"
+	"github.com/vladopajic/go-actor/actor"
```

Comment:
I'm going to have to spend some more time looking into the concurrency model here. Any reasons why you chose the actor model? 

I worked with teams that introduced actors only to get rid of them a year later because it became impossible to follow and optimise and very hard to test. So I'd like to be careful with that or make it easy to replace. The `go-actor` seems like a clean interface, but I haven't seen it used in other projects yet.

Reply:
Cool! I do see that it's quite lightweight. Some high-level idea about how actors are scheduled and what is blocking, what has a buffer would be a good idea to document. But I think I'm happy with the library choice.